pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -429,6 +429,11 @@ type WorkflowSpec struct {
 	// ArtifactGC describes the strategy to use when deleting artifacts from completed or deleted workflows (applies to all output Artifacts
 	// unless Artifact.ArtifactGC is specified, which overrides this)
 	ArtifactGC *WorkflowLevelArtifactGC `json:"artifactGC,omitempty" protobuf:"bytes,43,opt,name=artifactGC"`
+
+	// NodeName is a selector which will result in all pods of the workflow
+	// to be scheduled on the selected node(s). This is able to be overridden by
+	// a nodeName specified in the template.
+	NodeName string `json:"nodeName,omitempty" protobuf:"bytes,44,opt,name=nodeName"`
```

Comment:
this would make sense to put next to `nodeSelector` so that related fields are grouped together (also makes sense alphabetically)

Reply:
Done. Initially, I wanted to keep them sorted based on the proto tag, so adding them would be convenient in the future, but I also see an issue that aims to drop support for proto, so this doesn't hold value anymore.

Reply:
Oh, the proto number is technically relevant, though I don't think this is the first time we've added a field in the middle.
And yes #13542 is likely an upcoming change